chill-protobuf/src/main/java/com/twitter/chill/protobuf/ProtobufSerializer.java

```diff
@@ -69,6 +69,9 @@ public void write(Kryo kryo, Output output, Message mes) {
   public Message read(Kryo kryo, Input input, Class<Message> pbClass) {
     try {
       int size = input.readInt(true);
+      if (size == 0) {
+        return (Message) pbClass.getMethod("getDefaultInstance").invoke(null);
```

Comment:
do we actually need to use reflection here?

Should we cache the method in private variable so we don't need to make this call on each default value?

Reply:
1. Yes, we do as the `getDefaultInstance` method, like the `parseFrom` method is not defined on the `Message` abstraction.
2.  Good point. Might be better to cache it as well.

Reply:
Good point on caching! reflection is expensive.

One thing about though is that this inability to `parseFrom` empty byte array only happens with old versions of `protoc`.

Here we use a very old version so #571 updates it.

Reply:
Done with the cache changed.

@regadas thanks for #571